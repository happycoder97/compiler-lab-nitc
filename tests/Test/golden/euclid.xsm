0
2056
0
0
0
0
0
0
"":	XSM_MOV_Int SP 4096
"":	XSM_ADD_I SP 2
"":	XSM_UTJ (XSM_UTJ_CALL "main")
"":	XSM_INT 10
"ExtendedEuclid":	XSM_PUSH BP
"":	XSM_MOV_R BP SP
"":	XSM_ADD_I SP 8
"":	XSM_MOV_Int R0 4096
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_Int R1 5
"":	XSM_PUSH R1
"":	XSM_MOV_Int R1 (-2)
"":	XSM_PUSH R1
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_INT 7
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_MOV_Int R0 4097
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_Int R1 5
"":	XSM_PUSH R1
"":	XSM_MOV_Int R1 (-2)
"":	XSM_PUSH R1
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_INT 7
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_MOV_Int R0 0
"":	XSM_MOV_R R1 BP
"":	XSM_SUB_I R1 2
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_R SP BP
"":	XSM_POP BP
"":	XSM_RET
"":	XSM_MOV_Int R0 4096
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 (-4)
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_Int R0 4097
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 (-3)
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_Int R0 1
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 1
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_Int R0 0
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 2
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_Int R0 0
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 3
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_Int R0 1
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 4
"":	XSM_MOV_IndDst R1 R0
"L1":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 (-3)
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_Int R1 0
"":	XSM_NE R0 R1
"":	XSM_UTJ (XSM_UTJ_JZ R0 "L2")
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 (-4)
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 (-3)
"":	XSM_MOV_IndSrc R1 R1
"":	XSM_DIV R0 R1
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 5
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 (-4)
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 5
"":	XSM_MOV_IndSrc R1 R1
"":	XSM_MOV_R R2 BP
"":	XSM_ADD_I R2 (-3)
"":	XSM_MOV_IndSrc R2 R2
"":	XSM_MUL R1 R2
"":	XSM_SUB R0 R1
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 6
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 1
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 5
"":	XSM_MOV_IndSrc R1 R1
"":	XSM_MOV_R R2 BP
"":	XSM_ADD_I R2 2
"":	XSM_MOV_IndSrc R2 R2
"":	XSM_MUL R1 R2
"":	XSM_SUB R0 R1
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 7
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 3
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 5
"":	XSM_MOV_IndSrc R1 R1
"":	XSM_MOV_R R2 BP
"":	XSM_ADD_I R2 4
"":	XSM_MOV_IndSrc R2 R2
"":	XSM_MUL R1 R2
"":	XSM_SUB R0 R1
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 8
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 (-3)
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 (-4)
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 6
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 (-3)
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 2
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 1
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 7
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 2
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 4
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 3
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 8
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 4
"":	XSM_MOV_IndDst R1 R0
"":	XSM_UTJ (XSM_UTJ_JMP "L1")
"L2":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 (-4)
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_Int R1 5
"":	XSM_PUSH R1
"":	XSM_MOV_Int R1 (-2)
"":	XSM_PUSH R1
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_INT 7
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 1
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_Int R1 5
"":	XSM_PUSH R1
"":	XSM_MOV_Int R1 (-2)
"":	XSM_PUSH R1
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_INT 7
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 3
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_MOV_Int R1 5
"":	XSM_PUSH R1
"":	XSM_MOV_Int R1 (-2)
"":	XSM_PUSH R1
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_INT 7
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_MOV_Int R0 0
"":	XSM_MOV_R R1 BP
"":	XSM_SUB_I R1 2
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_R SP BP
"":	XSM_POP BP
"":	XSM_RET
"":	XSM_MOV_R SP BP
"":	XSM_POP BP
"":	XSM_RET
"main":	XSM_PUSH BP
"":	XSM_MOV_R BP SP
"":	XSM_ADD_I SP 3
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 1
"":	XSM_MOV_Int R1 7
"":	XSM_PUSH R1
"":	XSM_MOV_Int R1 (-1)
"":	XSM_PUSH R1
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_INT 6
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 2
"":	XSM_MOV_Int R1 7
"":	XSM_PUSH R1
"":	XSM_MOV_Int R1 (-1)
"":	XSM_PUSH R1
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_INT 6
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 1
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_PUSH R0
"":	XSM_MOV_R R0 BP
"":	XSM_ADD_I R0 2
"":	XSM_MOV_IndSrc R0 R0
"":	XSM_PUSH R0
"":	XSM_PUSH R0
"":	XSM_UTJ (XSM_UTJ_CALL "ExtendedEuclid")
"":	XSM_POP R0
"":	XSM_POP R1
"":	XSM_POP R1
"":	XSM_MOV_R R1 BP
"":	XSM_ADD_I R1 3
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_Int R0 0
"":	XSM_MOV_R R1 BP
"":	XSM_SUB_I R1 2
"":	XSM_MOV_IndDst R1 R0
"":	XSM_MOV_R SP BP
"":	XSM_POP BP
"":	XSM_RET
"":	XSM_MOV_R SP BP
"":	XSM_POP BP
"":	XSM_RET
