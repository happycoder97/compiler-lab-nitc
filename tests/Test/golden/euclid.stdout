29
8
-17
